src/main/java/teammates/ui/webapi/action/SubmitFeedbackResponsesAction.java

```diff
@@ -0,0 +1,221 @@
+                feedbackResponse.giver = giverIdentifier;
+                feedbackResponse.giverSection = giverSection;
+                feedbackResponse.recipient = recipient;
+                feedbackResponse.recipientSection =
+                        getRecipientSection(feedbackQuestion.getCourseId(), feedbackQuestion.getGiverType(),
+                                feedbackQuestion.getRecipientType(), recipient);
+                feedbackResponse.responseDetails = responseDetails;
+
+                feedbackResponsesToValidate.add(feedbackResponse);
+                feedbackResponsesToUpdate.add(FeedbackResponseAttributes.updateOptionsBuilder(feedbackResponse.getId())
+                        .withGiver(feedbackResponse.giver)
+                        .withGiverSection(feedbackResponse.giverSection)
+                        .withRecipient(feedbackResponse.recipient)
+                        .withRecipientSection(feedbackResponse.recipientSection)
+                        .withResponseDetails(feedbackResponse.getResponseDetails())
+                        .build());
+            } else {
+                FeedbackResponseAttributes feedbackResponse = FeedbackResponseAttributes
+                        .builder(feedbackQuestion.getId(), giverIdentifier, recipient)
+                        .withGiverSection(giverSection)
+                        .withRecipientSection(getRecipientSection(feedbackQuestion.getCourseId(),
+                                feedbackQuestion.getGiverType(),
+                                feedbackQuestion.getRecipientType(), recipient))
+                        .withCourseId(feedbackQuestion.getCourseId())
+                        .withFeedbackSessionName(feedbackQuestion.getFeedbackSessionName())
+                        .withResponseDetails(responseDetails)
+                        .build();
+
+                feedbackResponsesToValidate.add(feedbackResponse);
+                feedbackResponsesToAdd.add(feedbackResponse);
+            }
+        });
+
+        validateResponsesOfQuestion(feedbackQuestion, feedbackResponsesToValidate);
+
+        List<FeedbackResponseAttributes> feedbackResponsesToDelete = existingResponsesPerRecipient.entrySet().stream()
+                .filter(entry -> !responseDetailsPerRecipient.containsKey(entry.getKey()))
+                .map(entry -> entry.getValue())
+                .collect(Collectors.toList());
+
+        for (FeedbackResponseAttributes feedbackResponse : feedbackResponsesToDelete) {
+            logic.deleteFeedbackResponseCascade(feedbackResponse.getId());
+        }
+
+        List<FeedbackResponseAttributes> output = new ArrayList<>();
+
+        for (FeedbackResponseAttributes feedbackResponse : feedbackResponsesToAdd) {
+            try {
+                FeedbackResponseAttributes createdFeedbackResponse = logic.createFeedbackResponse(feedbackResponse);
+                createdFeedbackResponse.setId(
```

Comment:
Why we need set the ID manually? The ID should be updated correctly in the returned response. I believe the previous API is written before my improvement on DB layer.